### PR TITLE
Adding in Wrap functions for FSDP from Fairscale

### PR DESCRIPTION
**Pull request by sisilmehta2000**
Summary: as title

Test Plan: buck test mode/dev-nosan //caffe2/test/distributed/fsdp:wrap --keep-going

Differential Revision: D31936404

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### CODE REVIEW

**Review by rohan-varma — COMMENTED** [review]
Looks good to me overall to land this as a private API at first to unblock and then formalize the API. Although still have the question re: commented out test.

@zhaojuanmao can you also take a look?

File: test/distributed/fsdp/test_wrap.py
Comment (zhaojuanmao) [inline test/distributed/fsdp/test_wrap.py:6]: nit: remove this

**Review by zhaojuanmao — COMMENTED** [review]
looks good, waiting for the tests to pass

**Review by rohan-varma — APPROVED** [review]
LGTM, please wait for the ROCM CI to clear before landing, thanks for this!

**Pull request closed without merging**
